### PR TITLE
Hughbranch

### PR DESCRIPTION
**Pull request by hughbagan**
Task 2 and Task 5. I merged in the master branch earlier so it should be a straightforward merge.
I also added a new sql file for testing Task 5.
This solves issues #3 and #6

### CODE REVIEW

**Comment by hughbagan** [comment]
Upon reflection, the formatting for Task 2 might be a bit excessive. I think it's nice.

**Comment by NinjaPerson24119** [comment]
A few quick notes:
- Page switching API is inconsistent with Aida's (task 1)
- Upon picking a page, the info displays, but the API breaks

**Comment by hughbagan** [comment]
Nothing is breaking on my end; it still accepts input after a paper is selected. I can make it look like Aida's if you like. I just wrote it before then so that's why it's different.

File: src/main.py [hunk 1]
```diff
@@ -244,7 +368,48 @@ def task4(conn):
...
+    else:
+        pop = df
+    
+    print(pop)
+
+    # Display percentages in a pie chart
+    plot = pop.plot.pie(
+        labels=pop.area, 
+        y="count",
+        title="Most popular areas of expertise of papers",
```
Comment (NinjaPerson24119) [inline src/main.py]: "Most popular areas of expertise (by paper counts)"

File: src/main.py [hunk 2]
```diff
@@ -292,10 +457,10 @@ def main():
         print('What do you want to know?')
         print('[q] Quit')
         print('[1] Fetch all reviewers for a paper')
-        print('[2] Task 2')
+        print('[2] Find potential reviewers for papers')
         print('[3] Reviewers that have made [a, b] reviews')
         print('[4] Total number of sessions with accepted papers per author')
-        print('[5] Task 5')
+        print('[5] Find the most popular areas of expertise')
```
Comment (NinjaPerson24119) [inline src/main.py]: "Find the most popular areas of expertise (by paper counts)"

**Review by NinjaPerson24119 — COMMENTED** [review]
I'll admit, it's not a critical change, I'm just trying to be thorough. Only change it if you feel like it.
I'll post the error on my end in a sec.

**Comment by NinjaPerson24119** [comment]
More issues:
According to: https://eclass.srv.ualberta.ca/mod/forum/discuss.php?d=1140120
If there are ties, it should be possible to have more than five regions on the task5 pie display. I've tested with 7 regions all with 1 paper (7 ties), but only 5 regions are displayed. This is an error.

Honestly, it kind of pisses me off that the CS department of all people can't be bothered to make an unambiguous assignment spec.

**Comment by NinjaPerson24119** [comment]
More task5 issues:
I can select certain papers that aren't listed:
task5bug

**Comment by NinjaPerson24119** [comment]
On second thought, I think you should refactor to match Aida's interface. It's hard to read, especially for debugging as is. As well, the API doesn't always redisplay.

**Comment by NinjaPerson24119** [comment]
As well, it's a bit jarring that you expose the paper ID to the terminal. I think the papers should be listed like 1,2,3,4 not 1,5,16, etc. Task1 seems to correctly map the listed number to the actual page ID. Say, paper ID is 5, but the paper is the first in the list (1), typing 1 still refs the paper with ID=5

**Comment by hughbagan** [comment]
I honestly don't know how it managed to display papers 1, 3, 7, 9, 10 on one page. Even with my modified database it is working as expected.
Screenshot from 2019-03-09 18-09-32
Screenshot from 2019-03-09 18-09-48

**Comment by hughbagan** [comment]
I can redo the interface, but the selection should be working as intended.

**Comment by NinjaPerson24119** [comment]
I'm not using your DB. I modified it to have 7 regions each with one paper so I could test the tie breaking. E.g. some of the primary keys are missing because I didn't update them. We shouldn't rely on the primary keys being ordered anyway, since pages can be deleted and added at random. Idk why task1 is resilient to it, but I'll take a look.

Don't take the review personally btw, software never works as intended. That's why I disabled pushing straight to master / merging to master without a review. Hell, in the event that we have slightly different Python versions, that'd be enough to break something.

**Comment by NinjaPerson24119** [comment]
task2bug
Task2 has a bug where after I select a paper, the API doesn't redraw, but I can select another paper. I think it should go back to the main API like task1

**Comment by hughbagan** [comment]
I actually made it that way on purpose, so I'll make the necessary changes.

**Comment by NinjaPerson24119** [comment]
I looked over the SQL for task2 and it seems really minimal. Are you sure you can't do the query logic in SQL instead of python? If you can, we'll probably lose marks.

**Comment by hughbagan** [comment]
Sure, I'll see what I can do.

**Comment by hughbagan** [comment]
I see; I didn't understand before that the IDs could be so out of order. Aida's approach seems much better now. That's what happens when you code at night! Thanks :+1:

Task 2 has been rewritten, I'll push it when I'm done fixing Task 5

**Comment by hughbagan** [comment]
Changes have been pushed.

**Comment by NinjaPerson24119** [comment]
retake
Your tie breaker algorithm doesn't seem to work. Maybe sort the regions by popularity, take the first 5 of that sorted list, then if there are also regions equal to that 5th / last of the sorted list, add them to the graph.

**Comment by NinjaPerson24119** [comment]
Everything else looks good to me. Thanks for adding exceptions to the file opening in your code btw. Just fix the typo in task2, and fix the task5 tie breaker algorithm.

**Comment by hughbagan** [comment]
Looks like I just needed to decrease the bound by 1. Good catch :+1:
Where is the typo?

**Comment by NinjaPerson24119** [comment]
Check the above comments. You write num_pages instead of num_papers in task2. The result is that when the last page is selected, a number can't be entered that's greater than the number of pages.

**Comment by NinjaPerson24119** [comment]
Tested task5. lgtm.

**Review by aradu12 — APPROVED** [review]

**Pull request merged**